x-pack/plugin/search-engines/src/main/java/org/elasticsearch/searchengines/action/CreateSearchEngineAction.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.searchengines.action;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.ValidateActions;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.metadata.MetadataCreateIndexService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.util.CollectionUtils;
+import org.elasticsearch.indices.InvalidEngineNameException;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class CreateSearchEngineAction extends ActionType<AcknowledgedResponse> {
```

Comment:
It's great to have REST actions already implemented for engines, but shouldn't this be a part of [other task](https://github.com/elastic/enterprise-search-team/issues/2895)? It might have made this PR easier to digest (there's a lot going on here!) and it could have been used to onboard some other team members.

Reply:
Initially, I had only a create action for end-to-end testing. But @j-bennet was too fast on https://github.com/elastic/enterprise-search-team/issues/2895